torch/csrc/distributed/autograd/context/dist_autograd_container.cpp

```diff
@@ -128,12 +134,27 @@ DistAutogradContext& DistAutogradContainer::currentContext() {
   return it->second;
 }
 
-void DistAutogradContainer::releaseContext(int64_t context_id) {
+void DistAutogradContainer::releaseContext(
+    int64_t context_id,
+    bool notifyWorkers) {
```

Comment:
In which case would `notifyWorkers` be false?

Reply:
Right now, this is here since this PR doesn't support cycle detection, so this is a simple flag to avoid cycles. In a follow up PR I was planning to add additional changes to supported nested RPCs. 

More generally, there may be cases where we know that we don't want to identify additional workers (for example if we know all of them have already cleaned up the context), then we can avoid extra RPCs with this flag.

Reply:
Since we now have `releaseContextIfPresent`, `notifyWorkers` should no longer be needed since the RPC would just do a no-op if the context didn't exist, so we can get rid of `notifyWorkers`.